### PR TITLE
[SPARK-41822][CONNECT][TESTS][FOLLOW-UP] Remove the need of a fixed port to allow parallel running of tests

### PR DESCRIPTION
**Pull request by zhenlineo**
### What changes were proposed in this pull request?
Reusing the same fixed server port in the tests causes some flakiness of running tests in parallel. Especially on some systems that it is slow to recycle ports.

This PR makes the test to not use a fixed port and also to not start a connection unless it is needed.

### Why are the changes needed?
Small test reliability improvements. 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Only test changes.

### CODE REVIEW

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala [hunk 1]
```diff
@@ -133,11 +136,17 @@ class SparkConnectClientSuite extends ConnectFunSuite with BeforeAndAfterEach {
     TestPackURI(
       "sc://localhost:1234/",
       isCorrect = true,
-      client => testClientConnection(1234)(_ => client)),
+      client => {
```
Comment (zhenlineo) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:139]: @vicennial I removed the code to create a real connection as we've got it covered by

https://github.com/apache/spark/blob/cd6aa5819f3e97614532ec1f8c8d03b2a78c9340/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala#L91-L95

**Comment by zhenlineo** [comment]
@vicennial Can you review this PR? Thanks.

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala (hunk 1 above)
Comment (vicennial) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:139]: Can we also add a port-less connection string test?

Comment (vicennial) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:139]: LGTM otherwise!

Comment (zhenlineo) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala:139]: I kept the port-less one to still start the server as it is kind of important use case. So all good! See a few lines above the test.

https://github.com/apache/spark/blob/cd6aa5819f3e97614532ec1f8c8d03b2a78c9340/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala#L129-L132

**Review by vicennial — APPROVED** [review]
LGTM 🚢

**Comment by zhenlineo** [comment]
cc @HyukjinKwon @LuciferYang Can you have this tiny PR merged? Thanks?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**